addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/ExternalRedirectScanRule.java

```diff
@@ -294,8 +338,8 @@ private String getRefreshUrl(String value) {
     private boolean checkPayload(String value, String payload) {
         // Check both the payload and the standard url format
         return (value != null)
-                && (StringUtils.startsWithIgnoreCase(value, payload)
-                        || StringUtils.startsWithIgnoreCase(value, "http://" + REDIRECT_SITE));
+                && (StringUtils.containsIgnoreCase(value, payload)
+                        && (StringUtils.startsWithIgnoreCase(value, HttpHeader.HTTP)));
```

Comment:
- Why is it now requiring HTTP and no longer checking the site? Previous comment is now outdated.
 - Extra parentheses should be removed.

Reply:
First bullet already addressed per https://github.com/zaproxy/zap-extensions/pull/3858#discussion_r930885248

Reply:
Re-added site check. Tweaked parentheses. Relative cases do seem to apply with the JS redirects.